src/NLog.Targets.Syslog/SyslogTarget.cs

```diff
@@ -75,35 +75,47 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             Task.WhenAll(tasks)
                 .ContinueWith(t =>
                 {
-                    InternalLogger.Debug("[Syslog] Explicit flush completed");
+                    InternalLogger.Debug(t.Exception?.GetBaseException(), "[Syslog] Explicit flush completed");
                     asyncContinuation(t.Exception?.GetBaseException());
-                })
-                .Wait();
+                });
+        }
+
+        /// <inheritdoc />
+        protected override void CloseTarget()
+        {
+            DisposeDependencies();
+            base.CloseTarget();
         }
 
         /// <summary>Disposes the instance</summary>
         protected override void Dispose(bool disposing)
         {
             if (disposing)
             {
-                Enforcement.PropertyChanged -= Init;
-                MessageCreation.PropertyChanged -= Init;
-                MessageSend.PropertyChanged -= Init;
                 DisposeDependencies();
             }
             base.Dispose(disposing);
```

Comment:
Decompiling I saw that this calls CloseTarget that is said to close the target and release any unmanaged resources.
Since I do not have unmanaged resources cannot just I override dispose?

Reply:
Think the comment for CloseTarget was written in another time. Just like the comment for `IDisposable.Dispose` is misleading:

> Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.

Reply:
Ok but apart from comments: does it make any sense to override both?
Are there NLog parts in which CloseTarget is called instead of Dispose?

Reply:
I can just repeat what I have said before. Use `InitializeTarget` and `CloseTarget`. The Dispose-method is by default only called at end of application-lifetime (one time event). But `InitializeTarget` and `CloseTarget` can be called multiple times.

Reply:
Ok replacing my Dispose with CloseTarget I am compliant and produce the same effect

Reply:
Is there something called only once like Dispose but on initialization?

Reply:
The constructor is called at object-lifetime-start to prepare initial state. The dispose is called at application-lifetime-end.

Reply:
The problem is I would like to subscribe/unsubscribe from events only once and constructor/Dispose allow me to do so, but if I subscribe to events in the constructor, the initial configuration triggers event handlers

Reply:
Think you have the details you need to build something that will work. Maybe hook-up event-handlers in constructor, but have a Start / Stop-method to control whether event-handlers should be used. But the most important part is not to block in the FlushAsync-method :)